auth/providers/azure/pop_tokenverifier.go

```diff
@@ -210,6 +243,28 @@ func (p *PoPTokenVerifier) ValidatePopToken(token string) (string, error) {
 		return "", errors.Errorf("RSA verify err: %+v", err)
 	}
 
+	// Verify host 'nonce' claim
+	var nonce string
+	if nonceClaim, ok := claims["nonce"]; ok {
+		if _, ok := nonceClaim.(string); !ok {
+			return "", errors.Errorf("Invalid token. 'nonce' claim should be of type string")
+		}
+		nonce = nonceClaim.(string)
+	} else {
+		return "", errors.Errorf("Invalid token. 'nonce' claim is missing")
+	}
+	// Making sure nonce is not reused
+	err = nonceMap.AddToCache(nonce)
+	if err != nil {
+		return "", errors.Errorf("Invalid token. 'nonce' claim is reused")
+	}
+	klog.V(6).Infof("nonce claim added to the cache. Cache size is: %d", nonceMap.GetCounter())
+	// Cleaning cached nonce token after PoPTokenValidityDuration minutes + 1 minute by default
+	go func() {
```

Comment:
there is already a similar implementation that is uses a more efficient library. can we just do the same?
https://github.com/kubeguard/guard/blob/master/authz/providers/azure/data/datastore.go

Reply:
for instance, this implementation could probably use some jitter that's already implemented in those libraries

Reply:
Refactored to make it re-usable and leveraging it